api/configs/middleware/__init__.py

```diff
@@ -171,7 +171,7 @@ def SQLALCHEMY_DATABASE_URI(self) -> str:
 
     RETRIEVAL_SERVICE_EXECUTORS: NonNegativeInt = Field(
         description="Number of processes for the retrieval service, default to CPU cores.",
-        default=os.cpu_count(),
+        default_factory=lambda: NonNegativeInt(str(os.cpu_count())),
```

Comment:
`NonNegativeInt` is defined as an annotated form:

```python
NonNegativeInt = Annotated[int, annotated_types.Ge(0)]
```

So in theory this shouldn't be necessary?

Reply:
Yes, but all the changes are minimum required changes for passing the type checks.

Reply:
Oh, I checked again the errors from pymy. Changed to use `default=os.cpu_count() or 4,` in this case. The violation does not come from type wrapper , but form the optional None from `os_count()` instead. Thanks!